src/main/java/org/folio/rest/impl/InvoiceStorageImpl.java

```diff
@@ -105,9 +152,141 @@ public void deleteInvoiceStorageInvoiceLinesById(String id, String lang, Map<Str
   public void getInvoiceStorageInvoiceNumber(String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
     asyncResultHandler.handle(succeededFuture(Response.status(501).build()));
   }
+  
+  /**
+   * Creates a new sequence within a scope of transaction 
+   *
+   * @param Tx<Invoice>
+   *          Transaction of type Invoice
+   * @return Future of Invoice transaction with Invoice Line number sequence
+   */
+  private Future<Tx<Invoice>> createSequence(Tx<Invoice> tx) {
+    Future<Tx<Invoice>> future = Future.future();
 
-  @Override
-  public void getInvoiceStorageInvoiceLineNumber(String invoiceId, String lang, Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    asyncResultHandler.handle(succeededFuture(Response.status(501).build()));
+    String invoiceId = tx.getEntity().getId();
+    log.debug("Creating IL number sequence for invoice with id={}", invoiceId);
+    try {
+      pgClient.execute(tx.getConnection(), CREATE_SEQUENCE.getQuery(invoiceId), reply -> {
+        if (reply.failed()) {
+          log.error("IL number sequence creation for invoice with id={} failed", reply.cause(), invoiceId);
+          pgClient.rollbackTx(tx.getConnection(), rb -> future.fail(new HttpStatusException(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), reply.cause().getMessage())));
+        } else {
+          log.debug("IL number sequence for invoice with id={} successfully created", invoiceId);
+          future.complete(tx);
+        }
+      });
+    } catch (Exception e) {
+      future.fail(new HttpStatusException(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), e.getMessage()));
+    }
+    return future;
+  }
+
+  /**
+   * Deletes a sequence associated with the Invoice
+   *
+   * @param Invoice
+   *          Invoice with the sequence number to be deleted 
+   */
+  private void deleteSequence(Invoice invoice) {
+    if(invoice.getStatus() == Invoice.Status.CANCELLED || invoice.getStatus() == Invoice.Status.PAID) {
+      // Try to drop sequence for the IL number but ignore failures
+    	log.debug("InvoiceStorageImpl deleteSequence Drop sequence query -- ", DROP_SEQUENCE.getQuery(invoice.getId()));
```

Comment:
Better to assign `DROP_SEQUENCE.getQuery(invoice.getId())` to local variable because it is used 2 times and use parametrized logging message...

Reply:
Done